app/components/viral/modal_component.html.erb

```diff
@@ -0,0 +1,99 @@
+<div data-controller="viral--modal-component">
+  <!-- Modal toggle -->
+  <button
```

Comment:
**issue (blocking):** This current implementation will leave an button on the page with no text.  What could possibly work better, would be to accept two new attribute `trigger_text` and `trigger_tag` with the `trigger_tag` defaulting to `button`.  If the `trigger_text` is `nil` do not render the trigger (e.g. could be when triggering a turbo_frame render.  Use the `Viral::BaseComponent` to render the `trigger_tag` adding any `system_arguments` passed along, don't forget you will need to add the data attributes required to trigger the modal.

Reply:
It does leave an empty button on the page.  Please see the attached image.

![](https://github.com/phac-nml/irida-next/assets/11295750/d9a48af7-604e-484f-8b7a-2cda969cf3e1)

Reply:
Yes, that is hidden, not shown. As does clipboard.

Reply:
If you do not like `system_arguments` then you can do the same thing buy passing another argument, perhaps `trigger_classes` and pass the style that way.

Reply:
> Yes, that is hidden, not shown. As does clipboard.

The difference is that the clipboard component will throw an error if the label is not set (I just tested it), which is because it needs the button to actually work.  Here we do not need to have a button, so that it can work work turbo\_streams easily.  A button with no text is not "hidden" is is still in the document flow, which means that it could possibly be access via tabbing.

Reply:
I don't feel like it's necessary to accept 2 more attributes, when we know it should be triggered via a button if a label is passed. Maybe we can talk tomorrow.

Reply:
`**system_arguments` is a common pattern employed by both [Primer View Components](https://github.com/primer/view_components/blob/main/app/components/primer/base_component.rb) and [Polaris View Components](https://github.com/baoagency/polaris_view_components/blob/main/app/components/polaris/base_component.rb), which were both used as inspiration for our Viral View Components. It is basically a way of accepting unknown keyed arguments. If you take a look at `app/components/viral/card_component.rb` it accepts `**system_arguments` and merges the default classes with any provided classes. Then if you look at `app/components/viral/card_component.html.erb` it renders the `Viral::BaseComponent` passing the system arguments, in this case this results in a `section` tag with the specified `classes`.

With that all being said, for this PR I would like the button to be removed and instead just have the modal render open by default, which suits the purpose of your other PR. We can later revisit the best way to have a button launch the modal.

Reply:
Both potential solutions of a partial and button within component reverted.